src/adapter/src/catalog/migrate.rs

```diff
@@ -163,6 +175,314 @@ pub(crate) async fn migrate(
+                                    );
+                                    if table_ignore_columns.len() > 0 {
+                                        node.with_options.push(CreateSubsourceOption {
+                                            name: CreateSubsourceOptionName::IgnoreColumns,
+                                            value: Some(WithOptionValue::Sequence(
+                                                table_ignore_columns,
+                                            )),
+                                        });
+                                    }
+                                }
+                            }
+                            CreateSourceConnection::LoadGenerator { .. } => {
+                                // Load generator sources don't have any information we need to copy
+                                // to the subsource, so we just create an empty details for the subsource
+                                // which is used as a stub to ensure conformity with other source-export subsources
+                                let subsource_details = ProtoSourceExportStatementDetails {
+                                    kind: Some(
+                                        proto_source_export_statement_details::Kind::Loadgen(
+                                            ProtoLoadGeneratorSourceExportStatementDetails {},
+                                        ),
+                                    ),
+                                };
+                                node.with_options.push(CreateSubsourceOption {
+                                    name: CreateSubsourceOptionName::Details,
+                                    value: Some(WithOptionValue::Value(Value::String(
+                                        hex::encode(subsource_details.encode_to_vec()),
+                                    ))),
+                                });
+                            }
+                            _ => unreachable!("unexpected source type for existing subsource"),
+                        },
+                        _ => unreachable!("source must be a source"),
+                    };
+                    info!("migrated subsource: {:?}", node);
+                }
+            }
+        }
+    }
+
+    impl Rewriter<'_, '_> {
+        fn columns_for_table(
+            &self,
+            external_schema: &str,
+            external_table: &str,
+            all_columns: &Option<WithOptionValue<Raw>>,
+        ) -> Vec<WithOptionValue<Raw>> {
+            let all_table_columns = match all_columns {
+                Some(WithOptionValue::Sequence(columns)) => columns
+                    .into_iter()
+                    .map(|column| match column {
+                        WithOptionValue::UnresolvedItemName(name) => name,
+                        _ => unreachable!("text columns must be UnresolvedItemName"),
+                    })
+                    .collect::<Vec<_>>(),
```

Comment:
nit: You can skip the `collect` here and the corresponding `into_iter` below since the iterator chain just continues